osu.Game/Overlays/Settings/SettingsItem.cs

```diff
@@ -108,6 +126,22 @@ public virtual Bindable<T> Current
 
         public event Action SettingChanged;
 
+        /// <summary>
+        /// An action which when invoked will apply a classic default value to this setting.
+        /// </summary>
+        public Func<T> GetClassicDefault { get; set; }
```

Comment:
Has to be a function to allow for nullability (`T` may be a `class` or `struct` which is a bit limiting). Might be a better way around this that I'm missing.

Reply:
One alternative I can see is a property:
```csharp
        private T classicDefault;
        private bool hasClassicDefault;

        /// <summary>
        /// A "classic" default value for this setting.
        /// </summary>
        public T ClassicDefault
        {
            set => classicDefault = value;
        }

        public void ApplyClassicDefault(bool useClassicDefault)
        {
            if (hasClassicDefault && useClassicDefault)
                Current.Value = classicDefault;
            else
                Current.SetDefault();
        }
```

Reply:
@frenzibyte you forgot to set `hasClassicDefault` anywhere in that snippet i think?

Reply:
Property probably feels better, will change to that.